### ISSUE TITLE
Feature Request: Ability to override latitude

### ISSUE THREAD

**Issue by hoffbaked** [basnijholt/adaptive-lighting#97]
It would be helpful to be able to specify a latitude closer to the equator than your actual location. The places that get dark depressingly early in the Winter, also stay light really late in the Summer, so a fixed time offset isn't helpful.

**Comment by RubenKelevra** [comment]
Hey @hoffbaked,

I use fixed times for sunset and sunrise, which works pretty great for winter - is there a reason you don't want to use fixed times?

I guess a field to extend the daytime by x hours/minutes would be a bit more user-friendly, than entering coordinates - would this work for you as well?

**Comment by hoffbaked** [comment]
I guess the main thing is I would like it to follow the local seasonality somewhat, but not quite as drastically. And so far I’ve had to set up 9 separate adaptive lighting instances, since different areas and bulbs have different brightness ranges and such. It would be nice to not have to keep manually changing a bunch of parameters throughout the year in 9 different places.

Maybe an easier setting would be a simple percentage seasonality slider where 0% is following the sun at the equator, and 100% is local timing?

**Comment by RubenKelevra** [comment]
I don't think you would want to overwrite the time it takes for the sun to set - which is much shorter nearer to the equator than you might be used to.

How about just having a minimum time and a maximum time for sun set and sun rising, this way you can set it to something like 19:00 minimum and 22:30 maximum and this would limit the adaption.

**Comment by hoffbaked** [comment]
Hmmm. I’d never considered the difference in how slow or fast the sun sets. 

On the Summer Solstice in Seattle, the sun sets after 9PM, and the sky is fairly light until almost 11PM. In Winter, it’s about 6PM with the sun setting a little after 4PM. It would be nice to shave off an hour or two at the extremes. 

The min and max setting would work for that, but I’d rather have something that smoothly scales instead of clipping if it isn’t too hard.

**Comment by RubenKelevra** [comment]
> Hmmm. I’d never considered the difference in how slow or fast the sun sets. 
> 
> On the Summer Solstice in Seattle, the sun sets after 9PM, and the sky is fairly light until almost 11PM. In Winter, it’s about 6PM with the sun setting a little after 4PM. It would be nice to shave off an hour or two at the extremes. 
> 
> The min and max setting would work for that, but I’d rather have something that smoothly scales instead of clipping if it isn’t too hard. 

Well the curve itself won't clip, it would just stop moving past the min/max in the automatic mode where it shifts each day a little.

**Comment by gderber** [comment]
I was thinking about requesting a feature for min and max times for sunrise and sunset.  Glad to see a request is already here.

**Comment by mastermc0** [comment]
Just hit this one with winter in the UK would be good to have a max/min option so if sunset is earlier than the defined time that it uses this.

**Comment by RubenKelevra** [comment]
Hey guys, I wonder if it makes more sense to let users specify a "minimum amount of daylight", so Adaptive Lighting can just move the daylight in the center of the day and extend it past the sunrise/sunset?

Having a fixed time might be a bit less approachable since nobody really knows when the sun rises/sets just from memory.

**Comment by RubenKelevra** [comment]
Anyway, as soon as this has been determined it will be part of the v2 roadmap. :)

**Comment by gderber** [comment]
I think the min and max times would be the easiest for the user to understand.

min_sunrise_time: 6:00am
max_sunrise_time: 7:00am
min_sunset_time: 6:30pm
max_sunset_time: 8:00pm

Once that's set why would they need to remember when sunrise or sunset is from memory?

**Comment by GregWoods** [comment]
There are several issues, including this one, which can be fixed by allowing the setting of lat/long location, just as circadian lighting does.
Using a sunrise/sunset map of the world, you can pick a location has the exact offsets you need.
In the UK, I want a difference between summer and winter, but don't want the extremes of real life where it never gets fully dark in summer, and is only light for 6 hours in winter. I chose Lisbon, I tried it out using circadian lighting, and it works perfectly.
The problem is, circadian doesn't allow manually overriding the lights.

**Comment by basnijholt** [comment]
This is now fixed thanks to the work of @Hypfer in #358.

I have released Adaptive Lighting https://github.com/basnijholt/adaptive-lighting/releases/tag/1.2.0 which includes the change 😄
